### ISSUE TITLE
Place refactored notes in subfolders based on path in header #enhancement

### ISSUE THREAD

**Issue by dleeftink** [lynchjames/note-refactor-obsidian#51]
First, thank you for developing this plugin, it truly helps me to create more atomic notes. If development is still on-going, I would like to suggest the following: when refactoring a large note, I would like to organise refactored notes into subfolders based on header content. For instance:

```
# author/Smith, John
Some content about John Smith     [ creates a file called 'Smith, John.md' in the './author' folder ]

## author/Smith, John/document-a
Some document A notes             [ creates a file 'document-a.md' under the './author/Smith-John' folder ]

# author/Doe, Jane
Some content about Jane Doe       [ creates a file called 'Doe, Jane.md' in the  './author' folder ]

## author/Doe, Jane/document-b
Some document B notes             [ creates a file 'document-b.md' under the './author/Doe-Jane' folder ]
````

Default behaviour could be checking for whether folders exist and prompting the user to create one, or automatically create folders based on paths specified in headers. Additionally, folder and file names may be automatically sanitised/hyphenated. 

The main driver behind this request is that many authors/documents can be loaded sequentially into Obisidian, which can drive a network timelapse based on the sequence in which notes are refactored. This enables to see how various networks 'evolve'. While this can already be accomplished in part, Obsidian re-indexes files after manually moving them into subfolders, messing up the timing sequence. However, when notes are (automatically) placed into folders sequentially, the timing information stays intact.

**Comment by antomonte** [comment]
Hi,

assuming that the header already contains the **wikilink** to target containing the path starting from vault root, for instance...

[[path1/subpath1_1/Target_file]]

... yes, definitely I think that this can be considered a must for such great plugin.

**Comment by safrilth** [comment]
i agree.. this request can be considered a must for such great plugin...

it will enhance workflow for many people..
